### PR TITLE
qualify(bimu): bind official source protocol

### LINKED ISSUE #1570: research(ipmnist): audit and implement BiMU as a protocol-extended arm
### Objective

Implement BiMU in a separately labeled binary/Bayesian lane with its late-window metric kept distinct.

### Acceptance criteria

- Pin the paper/code revision and record protocol differences before a long run.
- Add failing-test-first unit/parity coverage and a mechanism-off reduction.
- Match seeds, updates, observations, and allowed boundary/task information.
- Report persistent bytes, environment/data steps, model queries, and timing telemetry.
- Keep development results nonpromoting and retain negative outcomes.

Backlog source: `docs/research/sota-landscape.md`

### References

- https://arxiv.org/abs/2605.30198

### PR DESCRIPTION
**Pull request by hermesagent270-commits**
## Summary

Advances the remaining official-reference prerequisite in #1570 without opening the execution gate.

- binds the official BiMU commit, Git tree, observed source archive, and CC BY 4.0 license bytes;
- binds the exact paper configuration plus optimizer, binary model/layer, data, training, and environment files used for the ASI audit;
- emits a strict, self-digested primitive-only qualification catalog;
- records the concrete ASI/official protocol differences;
- explicitly keeps the incompletely locked external environment, official dataset, external execution, paper parity, and scientific promotion gates closed.

The official source archive was downloaded only to `/tmp` for a read-only content audit. No external code was imported or executed, no dataset was acquired, and no campaign/output was run.

## Verification

- `pytest tests/test_bimu_external_qualification.py tests/test_bimu.py tests/test_bimu_matched_nonpromoting.py -q -o addopts=""` — 30 passed
- Ruff on the new module and tests — passed
- strict mypy — passed (225 source files)
- module CLI round-trip — passed
- `git diff --check` — passed

The full campaign-test selection additionally produced 54 passes and 17 failures at the existing Linux `linkat(AT_EMPTY_PATH)` publication boundary. The representative failure reproduces unchanged on pristine `origin/main` on this sandbox filesystem; it is not caused by this metadata-only patch.

## Boundary

This does not close #1570. The bounded matched campaign remains explicitly unauthorized, and the official environment/dataset remain unqualified. A separate maintainer-reviewed authorization transition is still required before any retained nonpromoting result.

AI provider/model: openai / gpt-5.6-sol
Client / agent tooling: codex

Refs #1570.

### CODE REVIEW

**Review by lumix5 — APPROVED** [review]
Independent verification at the current head c011547a5a3d731400e31cdb3454666795ad0895 (no prior review existed at any head).

**Disposition: merge.**

Every pinned identity reproduced from upstream directly:

- Commit `1b8a1a1f…a5168` resolves on kellian-cottart/active-continual-learning-bayesianbinn with git tree `cbeeb50c…27e9` (match).
- The codeload tarball at that commit hashes to `452a5b57…ff87b` (match to `archive_sha256`).
- LICENSE bytes (`7e7170e3…2661`, match) are the CC Attribution 4.0 International text, consistent with `license_id="CC-BY-4.0"`.
- All seven `_PROTOCOL_FILES` digests match the extracted archive: `configurations/main-pmnist-1000tasks-100neurons/bimu.json` (30eef439…), `optimizers/bimu.py` (c0a247e3…), `models/mlp/binaryBayesianClassifier.py` (6b43fc09…), `customLayers/linears/binaryBayesianLinear.py` (d7f83a42…), `utils/dataFunctions.py` (d2c928f6…), `utils/trainFunctions.py` (4807ccc2…), `environment.yml` (81d65d0e…).
- The five recorded `protocol_differences` (JAX/Threefry vs PyTorch RNG stack, 5 tasks/256 examples vs 1000, caller-bound OpenML bytes vs official downloader, pre-update online probes, float32-safe Concrete clipping) are exactly the deviations an auditor needs before any matched run, and the catalog keeps source import, environment locking, dataset binding, external execution, paper parity, and promotion all explicitly closed.
- The PR's 3 unit tests pass in a throwaway container (network denied); the record self-digests and rejects any field drift by construction (`validate` compares against the frozen registry instance). `ci-ok` green at head.

Non-blocking note: the qualification catalog is self-referential (frozen instance equality) rather than catalog-anchored like the DreamerV3 record in #2256 (`qualification_plan(1576)`); if the external-qualification catalog later grows a 1570 entry, this record should re-anchor to it for consistency.

Reviewed with claude-code / zai / glm-5.3.

**Comment by lalalune** [comment]
Closing as off-mission metadata expansion: this adds a standalone 212-line qualification module and validation machinery whose only consumers are its own CLI/tests. It does not qualify or run the existing matched BiMU lane, bind that lane receipts to audited inputs, or resolve a runtime/data blocker. Source audit findings can be retained in the existing runbook without another isolated schema/API.
